src/test/scala/com/databricks/spark/redshift/RedshiftSourceSuite.scala

```diff
@@ -0,0 +1,486 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift
+
+import java.io.File
+import java.sql.{Connection, PreparedStatement, SQLException}
+
+import scala.util.matching.Regex
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.mapreduce.InputFormat
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Matchers}
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.jdbc.JDBCWrapper
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.{Row, SQLContext, SaveMode}
+
+class TestContext extends SparkContext("local", "RedshiftSourceSuite") {
+
+  /**
+   * A text file containing fake unloaded Redshift data of all supported types
+   */
+  val testData = new File("src/test/resources/redshift_unload_data.txt").toURI.toString
+
+  override def newAPIHadoopFile[K, V, F <: InputFormat[K, V]]
+  (path: String, fClass: Class[F], kClass: Class[K],
+   vClass: Class[V], conf: Configuration = hadoopConfiguration):
+  RDD[(K, V)] = {
+    super.newAPIHadoopFile[K, V, F](testData, fClass, kClass, vClass, conf)
+  }
+}
+
+/**
+ * Tests main DataFrame loading and writing functionality
+ */
+class RedshiftSourceSuite
+  extends FunSuite
+  with Matchers
+  with MockFactory
+  with BeforeAndAfterAll {
+
+  /**
+   * Temporary folder for unloading data to
+   */
+  val tempDir = {
+    var dir = File.createTempFile("spark_redshift_tests", "")
+    dir.delete()
+    dir.mkdirs()
+    dir.toURI.toString
+  }
+
+  /**
+   * Expected parsed output corresponding to the output of testData.
+   */
+  val expectedData =
+    Array(
+      Row(1.toByte, true, TestUtils.toTimestamp(2015, 6, 1, 0, 0, 0), 1234152.123124981,
+        1.0f, 42, 1239012341823719L, 23, "Unicode是樂趣", TestUtils.toTimestamp(2015, 6, 1, 0, 0, 0, 1)),
+      Row(1.toByte, false, TestUtils.toTimestamp(2015, 6, 2, 0, 0, 0), 0.0, 0.0f, 42, 1239012341823719L, -13, "asdf",
+        TestUtils.toTimestamp(2015, 6, 2, 0, 0, 0, 0)),
+      Row(0.toByte, null, TestUtils.toTimestamp(2015, 6, 3, 0, 0, 0), 0.0, -1.0f, 4141214, 1239012341823719L, null, "f",
+        TestUtils.toTimestamp(2015, 6, 3, 0, 0, 0)),
+      Row(0.toByte, false, null, -1234152.123124981, 100000.0f, null, 1239012341823719L, 24, "___|_123", null),
+      Row(List.fill(10)(null): _*))
+
+
+  /**
+   * Spark Context with hadoop file overridden to point at our local test data file for this suite,
+   * no-matter what temp directory was generated and requested.
+   */
+  private var sc: SparkContext = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sc = new TestContext
+  }
+
+  override def afterAll(): Unit = {
+    val temp = new File(tempDir)
```

Comment:
Since we already depend on Guava, why not use one of its file helper methods to delete the non-empty test directory?